src/app/pages/registration/registration-form/registration-form.component.html

```diff
@@ -5,99 +5,74 @@ <h2>Registration Form</h2>
-
-      </div>
-      <div class="form-group required">
-        <label for="name">full name</label>
-        <input type="text" id="name" class="form-control" [(ngModel)]="model.name" #name="ngModel" name="name" required
-               maxlength="255">
-        <div class="alert alert-danger" role="alert" *ngIf="name.errors?.required && (name.touched || submitted)">
+    <form (ngSubmit)="save()" [formGroup]="registrationForm" *ngIf="registrationForm">
+      <mat-form-field class="full-width">
+        <input matInput placeholder="voucher" type="text" id="voucher2"
+             name="voucher" formControlName="voucher"
+             maxlength="255">
+      </mat-form-field>
+      <mat-form-field>
+        <input type="text" id="city" placeholder="city" formControlName="city"
+             matInput name="city"
+             required maxlength="255">
+        {{registrationForm['city']|json}}
+        <mat-error *ngIf="city.invalid && (city.dirty || city.touched || submitted)">
           This field is required
-        </div>
-      </div>
-      <div class="form-group required">
-        <label for="email">e-mail</label>
-        <input type="email" id="email" class="form-control" [(ngModel)]="model.email" #email="ngModel" name="email" required
-               maxlength="255" email>
-        <div class="alert alert-danger" role="alert" *ngIf="email.errors?.required && (email.touched || submitted)">
+        </mat-error>
+      </mat-form-field>
+      <label>sex*: </label>
+      <mat-radio-group formControlName="sex">
+        <mat-radio-button type="radio" name="sex" value="F" required> female</mat-radio-button>
+        <mat-radio-button type="radio" name="sex" value="M" required> male</mat-radio-button>
+        <mat-error *ngIf="sex.invalid && (sex.dirty || sex.touched || submitted)">
           This field is required
-        </div>
-        <div class="alert alert-danger" role="alert" *ngIf="email.errors?.email && (email.touched || submitted)">
-          This is not valid email
-        </div>
-      </div>
-      <div class="form-group required">
-        <label for="city">city</label>
-        <input type="text" id="city" class="form-control" [(ngModel)]="model.city" #city="ngModel" name="city" required
-               maxlength="255">
-        <div class="alert alert-danger" role="alert" *ngIf="city.errors?.required && (city.touched || submitted)">
+        </mat-error>
+      </mat-radio-group>
+      <mat-form-field>
+        <mat-select name="size" id="size" formControlName="size" required placeholder="t-shirt size">
+          <mat-option></mat-option>
+          <mat-option value="S">S</mat-option>
+          <mat-option value="M">M</mat-option>
+          <mat-option value="L">L</mat-option>
+          <mat-option value="XL">XL</mat-option>
+          <mat-option value="XXL">XXL</mat-option>
+        </mat-select>
+        <mat-error *ngIf="size.invalid && (size.dirty || size.touched || submitted)">
```

Comment:
those additional checks are not needed in angular Material Design. It has very sensible dedaults.

so we just need `*ngIf='side.invalid'`